gc/base/MarkMap.hpp

```diff
@@ -102,6 +102,19 @@ class MM_MarkMap : public MM_HeapMap
 		return _heapMapBaseDelta + (slotIndex << _heapMapIndexShift);
 	}
 
+	/*
+	 * check MarkMap if there is any liveObjects in the Card
+	 * this function assumes that card covers exactly 512 bytes.
+	 * @param heapAddress has to be card size aligned
+	 */
+	MMINLINE bool isAnyLiveObjectsInCard(void* heapAddress)
+	{
+		#if (8 != BITS_PER_BYTE) || (9 != CARD_SIZE_SHIFT)
+		#error Card size has to be exactly 512 bytes
+		#endif
+		uint64_t* map4Card = (uint64_t*) &((uintptr_t *)(getMarkBits()))[getSlotIndex((omrobjectptr_t) heapAddress)];
```

Comment:
I don't like the name `map4Card`, what about `cardmap`?

Reply:
I think this could be cleaner if we added some new helpers to the `MM_HeapMap` base class. Maybe something like:

```c++
MMINLINE uintptr_t*
getSlotPtr(size_t index) const
{
	return &getHeapMapBits()[index];
}

MMINLINE uintptr_t*
getSlotPtrForAddress(uintptr_t address) const
{
	return getSlotPtr(getSlotIndex(address));
}
```

Then this code can be:

```c++
	return 0 != *(uint64_t*)getSlotPtrForAddress(heapAddress);
```

What do you think?